src/java/org/apache/cassandra/io/util/SequentialWriter.java

```diff
@@ -177,50 +177,55 @@ public void skipBytes(int numBytes) throws IOException
 
     /**
      * Synchronize file contents with disk.
+     * <p/>
+     * This is only safe to call before truncation or close for CompressedSequentialWriter
+     * Otherwise it will leave a non-uniform size compressed block in the middle of the file
+     * and the compressed format can't handle that.
      */
     public void sync()
     {
-        syncInternal();
+        doFlush(true);
     }
 
-    protected void syncDataOnlyInternal()
+    private void syncInternal(boolean syncMetadata)
```

Comment:
I'd consider renaming the function. Before it was kind of API function called everywhere.
Now it's just a thin wrapper over `SyncUtils`.

I don't insist, but it took me a while to untangle what each function was doing before and what is doing now.